cmd/preflight/cmd/check_container.go

```diff
@@ -22,7 +24,10 @@ import (
 
 var submit bool
 
-func checkContainerCmd() *cobra.Command {
+// runPreflight is introduced to make testing of this command possible, it has the same method signature as cli.RunPreflight.
```

Comment:
I see you have defined `mockRunPreflight` in the test code to have the same signature as `cli.RunPreflight` which makes sense. Do you need another `type runPreflight` here ? Can't you just do `checkContainerCmd(runprelight cli.RunPreflight)` (I might be confused 😅)

Reply:
The type definition could be placed in the cli package though right ?

Reply:
@skattoju That's what I was trying to do in the first place, but then you end up with the problem of not being about to 'inject' it into the CMD. Also, since the CLI package is internal, it shouldn't care about how/why a `caller` needs to mock it's function call to be testable.